src/foam/mlang/mlangJava.js

```diff
@@ -567,7 +567,7 @@ for ( int i = 0; i < length; i++ ) {
   if ( obj[i] == null )
     builder.append("");
   else 
-    builder.append(obj[i]);
+    builder.append(obj[i].toString().replace("\\\\","\\\\\\\\").replace(",","\\\\,"));
```

Comment:
Is this regex correct? Does it escape \ as \\ or just \\ as \\\\?

In-line regexes like this are difficult to understand. Instead, make a helper method which describes what it does and then move the regex to that function. Something like: escapeCommas().

Even better, do something like escapeCommasAndAppend(builder, obj[i]) which does the escaping and appending in one step to avoid generating garbage, which will slow us down. Not such a big deal in this case anyway since Postgres is already slow, but generally that would be a better design.